[Graphics] Show Equipment Regardless of Age

Patrick12115 opened the pull request:
This PR was made to essentially be a fill in for whenever #3288 gets done. This is just a lighter weight, more simplified way to allow the equipment to be visible when using it as any age.

The hands are patched to display the correct age's hand, but only with vanilla models. Custom models will show the hand that is normally applied to the model.

The pause screen Link model is unchanged, as it is coded completely separately and felt like overkill to do with this PR.

Sheathes also don't show properly in all scenarios, couldn't figure out if that was even possible without more extensive code changes.

It is also best used with Disable LOD checked, as no "far" DL's are being patched for the hands, as I felt that would be more unnecessary code bloat for little gain.

Feedback is welcome.

![image](https://github.com/HarbourMasters/Shipwright/assets/115201185/49688278-5d80-4d0a-9402-93f3f4575c5b)
![image](https://github.com/HarbourMasters/Shipwright/assets/115201185/94359e5b-2e01-4e2c-a5fc-0aa34a41d704)
![image](https://github.com/HarbourMasters/Shipwright/assets/115201185/341c50ad-bd2f-44f7-8571-9b6e9cd3c5b7)
![image](https://github.com/HarbourMasters/Shipwright/assets/115201185/baaed352-b5e8-4f1b-a500-66986411d445)
![image](https://github.com/HarbourMasters/Shipwright/assets/115201185/a55df9ec-e704-4d89-bfc3-083c45905fd0)





<!--- section:artifacts:start -->
### Build Artifacts
  - [soh.otr.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/1210722909.zip)
  - [soh-linux-compatibility.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/1210722911.zip)
  - [soh-linux-performance.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/1210722912.zip)
  - [soh-mac.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/1210722913.zip)
  - [soh-switch.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/1210722915.zip)
  - [soh-wiiu.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/1210722917.zip)
  - [soh-windows.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/1210722918.zip)
<!--- section:artifacts:end -->

Patrick12115 committed:
Hookshot, Hammer, & Boomerang

Patrick12115 committed:
Added the remainder of equipment and DL Patching for hands

Patrick12115 committed:
Removed Mirror Shield Patch

Messed with the front texture of shield

aMannusAtePoop commented:
Sweet fix! Lemme ask you though. If you use a custom model for the swords, will both ages have it in hand?

Patrick12115 commented:
Dude, go away. Especially with that name.

TheInvincibleUrine commented:
Will you be mah fren, Patrick? I respect your choice to be aMannus's bed buddy. You are a good nonbinary LGBTQP+

TheInvincibleUrine commented:
Abuse? I praised his choice to be gay! Lol. No satisfying the Far-Left is there?

Patrick12115 committed:
Child Hylian Shield Cheat

Adds a cheat that allows Child Link to hold Hylian Shield as adult. Added to this since it uses the same way to render the shield

Patrick12115 commented:
Added Cheat for allowing Child Link to hold the Hylian Shield like Adult Link. Added under cheats due to the ability to skip logically needing Deku shield for Deku Tree. Also added to here, instead of it's own PR because it uses the same process to render the shield in Links hand.

Patrick12115 committed:
Debugging

Patrick12115 committed:
Adds Scaling to Child Link with Adult Equipment

Patrick12115 committed:
Merge remote-tracking branch 'upstream/develop' into Ageless-Items-Showing

Patrick12115 committed:
Fix Merge Conflict Blunders

Patrick12115 committed:
More cleanup

Patrick12115 committed:
Added Scaling checkbox and some clean up

Patrick12115 commented:
Added Scaling of Adult equipment for Child Link as a checkbox and added an option to rotate the Child Hylian shield with the cheat to hold it like Adult is on, making it the same orientation as the rest of the shields.

Patrick12115 committed:
Added an || ITEM_NONE

Patrick12115 committed:
More cleanup and simplification

Patrick12115 committed:
Accidently added spacer

Pepe20129 commented on soh/src/code/z_player_lib.c:
```diff
@@ -580,7 +580,11 @@ s32 func_8008E9C4(Player* this) {
 
 s32 Player_IsChildWithHylianShield(Player* this) {
-    return gSaveContext.linkAge != 0 && (this->currentShield == PLAYER_SHIELD_HYLIAN);
+    if (CVarGetInteger("gChildHoldsHylianShield", 0)) {
+		return 0;
+	} else {
+        return gSaveContext.linkAge != 0 && (this->currentShield == PLAYER_SHIELD_HYLIAN);
+    }
```
```suggestion
    if (CVarGetInteger("gChildHoldsHylianShield", 0)) {
        return 0;
    }
    return gSaveContext.linkAge != 0 && (this->currentShield == PLAYER_SHIELD_HYLIAN);
```

Pepe20129 commented on soh/src/overlays/actors/ovl_player_actor/z_player.c:
```diff
@@ -2911,7 +2911,11 @@ s32 func_808356E8(Player* this, PlayState* play) {
 
 void func_808357E8(Player* this, Gfx** dLists) {
-    this->leftHandDLists = &dLists[gSaveContext.linkAge];
+    if (LINK_IS_ADULT && (CVarGetInteger("gEquipmentAlwaysVisible", 0))) {
+		this->leftHandDLists = &dLists[1];
+	} else {
+		this->leftHandDLists = &dLists[gSaveContext.linkAge];
+	}
```
```suggestion
    if (LINK_IS_ADULT && (CVarGetInteger("gEquipmentAlwaysVisible", 0))) {
        this->leftHandDLists = &dLists[1];
    } else {
        this->leftHandDLists = &dLists[gSaveContext.linkAge];
    }
```

Pepe20129 reviewed (approved):
Not blocking but there's some weirdness with tabs instead of spaces

Patrick12115 committed:
Replace B_BTN_ITEM

Patrick12115 committed:
Spacing

Co-authored-by: Pepe20129 <72659707+Pepe20129@users.noreply.github.com>

Patrick12115 committed:
Added most sheath functionality

Patrick12115 committed:
Updated tooltips

Patrick12115 committed:
removed child hylian shield and tweaked bow/slingshot

Made child hylian shield cheat into a separate branch and hooked into existing bow/slingshot enhancement for drawing those

Patrick12115 commented:
Separated Child Hylian Shield cheat into it's own PR #3887

Patrick12115 committed:
Prefixes

Patrick12115 committed:
()

garrettjoecox committed:
Merge branch 'develop' into Ageless-Items-Showing
